### ISSUE TITLE
Investigate: File as a source

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/quickwit#1159]
We already have a file source but right now it is only good as a way to index a bunch of static files.

It would be great however to have a source that handles log rotation.

For instance, if a system does log rotation in a way that keeps files in order,
we could use whatever we want as a partition id and use 
`lastindexed_filename @ offset in file` as a position.

This ticket requires first investigating a bunch of system doing log rotation, seeing if there is a common denominator
that might be used to flesh out of a possible specification and its configuration.

**Comment by runiq** [comment]
Maybe just plug [Vector](https://vector.dev) as a means to ingest from files? It already handles log rotation (see [File source#How it works](https://vector.dev/docs/reference/configuration/sources/file/#how-it-works)).

**Comment by fmassot** [comment]
Yes, using Vector is relevant. But some users may want to do that without external components. By the way, I would rather focus on an S3 source, and vector already has it.
